### DISCUSSION TITLE
Copyright Assignment For Contributions

### DISCUSSION THREAD

**Discussion by donn** [The-OpenROAD-Project/OpenLane#776] [Ideas]
### Prompt
Currently, our copyrights are a bit of a mess. Situations such as #771 come to mind, where people don't know what the proper copyright assignment is.

### Proposal
I find @google's way of doing things uniquely sober: https://opensource.google/docs/releasing/authors/

Essentially, we are to replace the copyright headers with `The OpenLane Authors`. The file already has a link to a list of contributors and a list of major contributors.

@mithro  @mkkassem thoughts?

**Comment by maliberty** [comment]
The link you provide states "This information is in addition to go/releasing/contributions. It does not replace it.".   The go link requires a CLA which is rather onerous.

**Comment by donn** [comment]
That... is true. Gah.

Well I'm at a bit of a loss frankly, as the situation is currently a bit muddy: there are contributors who, despite legally having copyright over their changes, chose to not add their names to headers. So this whole thing is a bit of a gray area.

**Reply by tamood** [reply]
@donn, I suggest you to write a guideline on how to write a "proper copyright statement", as a maintainer of this project. I an unable to see any problem in the copyright declaration of libtrim.py that is consistent with libtrim.pl and the codebase, even from the perspective of https://opensource.google/docs/releasing/authors/.

If you see any inconsistency, you should comment in the PR review itself.

**Reply by donn** [reply]
Translations from one language to another are considered derivative work, of which you've removed Efabless's copyright. You're entitled to include yours, of course, but derivative works remain copyrighted by the original developer as well.

**Reply by tamood** [reply]
@donn, Now, I understand you point. Though not an expert of copyright, I hold that both pieces of codes are entirely different in their logic and implementation, so the contribution is in no way "derived" but equivalent. However, we may write something like "Originally written in Perl by ______ at eFabless Corp., and available as libtrim.pl for reference."

**Reply by donn** [reply]
If you do hold that, then I'll take you at your word. You don't need the notice. In any case, this discussion should be on your PR and not here.

**Comment by maliberty** [comment]
Fwiw in OR we require a DCO and have a github action to enforce that.  Note that is can be a pain when people forget.

**Reply by donn** [reply]
That actually could work. I'm used to painful PRs by now, so that's not an issue.
